src/rust/en.mangakatana/src/lib.rs

```diff
@@ -0,0 +1,186 @@
+#![no_std]
+
+mod helper;
+mod parser;
+
+use aidoku::{
+	error::Result,
+	prelude::*,
+	std::defaults::defaults_get,
+	std::net::{HttpMethod, Request},
+	std::{String, Vec},
+	Chapter, DeepLink, Filter, FilterType, Listing, Manga, MangaPageResult, Page,
+};
+
+use helper::*;
+use parser::*;
+
+const URL: &str = "https://mangakatana.com";
+
+#[get_manga_list]
+fn get_manga_list(filters: Vec<Filter>, page: i32) -> Result<MangaPageResult> {
+	// filter=1 enables filtering
+	let mut url = format!("{}/manga/page/{}?filter=1", URL, page);
+
+	let mut searching = false;
+	let mut include_generes = Vec::new();
+	let mut exclude_generes = Vec::new();
+
+	for filter in filters {
+		match filter.kind {
+			FilterType::Title => {
+				if let Ok(value) = filter.value.as_string() {
+					searching = true;
+					let query = urlencode(value.read());
+					url = format!("{}/page/{}?search={}&search_by=book_name", URL, page, query);
+				}
+			}
+			FilterType::Author => {
+				if let Ok(value) = filter.value.as_string() {
+					searching = true;
+					let query = urlencode(value.read());
+					url = format!("{}/page/{}?search={}&search_by=author", URL, page, query);
+				}
+			}
+			FilterType::Genre => {
+				if let Ok(id) = filter.object.get("id").as_string() {
+					match filter.value.as_int().unwrap_or(-1) {
+						0 => exclude_generes.push(id.read()),
+						1 => include_generes.push(id.read()),
+						_ => continue,
+					}
+				}
+			}
+			FilterType::Select => match filter.name.as_str() {
+				"Status" => match filter.value.as_int().unwrap_or(-1) {
+					0 => continue,
+					1 => url.push_str("&status=1"),
+					2 => url.push_str("&status=2"),
+					3 => url.push_str("&status=0"),
+					_ => continue,
+				},
+				"Genre Inclusion Mode" => match filter.value.as_int().unwrap_or(-1) {
+					0 => url.push_str("&include_mode=and"),
+					1 => url.push_str("&include_mode=or"),
+					_ => url.push_str("&include_mode=and"),
```

Comment:
probably best to `continue` in the default case (shouldn't be hit anyways though)
```suggestion
					_ => continue,
```

Reply:
I did this on purpose as it is the default for mangakatana, mangakatana is weird where if you are missing some arguments it throws everything else off. Same for the other places where I added default url params instead of continuing.

Reply:
hmm okay